frontend/gateway/grpcclient/client.go

```diff
@@ -105,22 +105,39 @@ func (c *grpcClient) Run(ctx context.Context, f client.BuildFunc) (retError erro
 					Metadata: res.Metadata,
 				}
 				if res.Refs != nil {
-					m := map[string]*pb.Ref{}
-					for k, r := range res.Refs {
-						id, err := convertRef(r)
-						if err != nil {
-							retError = err
-							continue
+					if c.caps.Supports(opspb.CapProtoRefArray) == nil {
```

Comment:
I think these are incompatible sets here. `c.caps` is gateway caps and `opspb.CapProtoRefArray` looks to be llb cap. Should be gateway cap.